build/filter_test.go

```diff
@@ -279,4 +281,635 @@ var _ = Describe("Test filters utilities:", func() {
 
 	})
 
+	Describe("Test  fileSystemFilter", func() {
+		It("", func() {
+			filter := &FileSystemFilter{}
+
+			// Create a test case with sample input
+			system := &v1.System{
+				Spec: v1.SystemSpec{
+					Storage: &v1.SystemStorageInfo{
+						FileSystems: &v1.ControllerFileSystemList{
+							{ // Adding file systems for testing
+								Name: "backup",
+								Size: 100,
+							},
+							{
+								Name: "database",
+								Size: 200,
+							},
+							{
+								Name: "instances",
+								Size: 300,
+							},
+							{
+								Name: "image-conversion",
+								Size: 400,
+							},
+							{
+								Name: "extra", // this will be filtered out
+								Size: 500,
+							},
+						},
+					},
+				},
+			}
+
+			deployment := &Deployment{} // Dummy deployment for testing, assuming you have a Deployment type
+
+			// Call the Filter method
+			err := filter.Filter(system, deployment)
+			Expect(err).To(BeNil())
+
+			// Check if file systems have been filtered as expected
+			expectedFilteredFileSystems := []v1.ControllerFileSystemInfo{
+				{
+					Name: "backup",
+					Size: 100,
+				},
+				{
+					Name: "database",
+					Size: 200,
+				},
+				{
+					Name: "instances",
+					Size: 300,
+				},
+				{
+					Name: "image-conversion",
+					Size: 400,
+				},
+			}
+			list := v1.ControllerFileSystemList(expectedFilteredFileSystems)
+			Expect(&list).To(Equal(system.Spec.Storage.FileSystems))
+
+		})
+	})
+
+	Describe("Test  CACertificateFilter", func() {
+		It("", func() {
+			filter := &CACertificateFilter{}
+
+			// Create a test case with sample input
+			system := &v1.System{
+				Spec: v1.SystemSpec{
+					Certificates: &v1.CertificateList{
+
+						{ // Adding certificate info for testing
+							Type: v1.PlatformCACertificate, // this will be filtered out
+						},
+						{
+							Type: v1.OpenstackCACertificate, // this will be filtered out
+						},
+						{
+							Type: v1.DockerCertificate,
+						},
+					},
+				},
+			}
+
+			deployment := &Deployment{} // Dummy deployment for testing, assuming you have a Deployment type
+
+			// Call the Filter method
+			err := filter.Filter(system, deployment)
+			Expect(err).To(BeNil())
+
+			// Check if file systems have been filtered as expected
+			expectedFilteredCertificates := []v1.CertificateInfo{
+				{
+					Type: v1.DockerCertificate,
+				},
+			}
+			list := v1.CertificateList(expectedFilteredCertificates)
+			Expect(&list).To(Equal(system.Spec.Certificates))
+
+		})
+
+	})
+
+	Describe("Test ServiceParameterFilter", func() {
```

Comment:
missing "Context" intentionally?
i think its good to have from readability perspective.

Reply:
Updated